docs/documentation/server_admin/topics/workflows/understanding-workflows-engine.adoc

```diff
@@ -0,0 +1,61 @@
+[id="understanding-workflow_{context}"]
+
+[[_understanding_workflows_engine_]]
+=  Understanding the workflows engine
+[role="_abstract"]
+
+The lifecycle and execution of workflows in {project_name} is managed by the Workflows Engine.
+The engine is responsible for processing events, creating workflow executions, and processing their steps.
+Once a workflow execution is created, the engine takes over and manages the execution of its steps chain.
+
+The execution of the steps chain is done asynchronously and detached from the request that originated the event that
+triggered the workflow. This means that workflows are not scheduled nor executed immediately within the context of the request,
+but rather sent to a task thread pool and scheduled to be executed later by the engine. This allows for better performance and scalability, as the steps can be executed
+in the background without blocking the main request thread.
+
+For immediate steps, the engine processes them as soon as the workflow execution is submitted to the task thread pool. This means that immediate steps are executed
+right away, as soon as the workflow execution is picked up by a worker thread from the pool.
+
+For scheduled steps, the engine continuously monitors the state of all active workflow executions in the realm. This is done by a background
+task that runs periodically to check for any workflow executions that have steps ready to be executed. The schedule steps of a workflow are also executed asynchronously
+by the background task, using the same task thread pool as immediate steps.
+
+== Configuring the scheduled steps execution interval
+
+By default, the background task tracking due steps runs every 12 hours, but this interval can be configured by setting the following server configuration option:
+
+*   `spi-events-listener--workflow-event-listener--step-runner-task-interval`: Defines the interval, in milliseconds, at which the background task runs to check for workflow steps that are due for execution.
+
+You can adjust this interval based on your realm's needs and the expected frequency of workflow executions.
```

Comment:
Should we allow setting this in the UI?  There might be times when an administrator wants to speed it up or slow it down without rebooting the server.

Also, it might need to be adjusted on a per-realm basis.

Reply:
I'll replace the `realm's needs` with `realms need` because the runner task is a server-level task. We can review that in the future and have a task for each realm where you can adjust its settings on a per-realm basis.